File: server/utils/spleeter_separation.py
import sys
from spleeter.separator import Separator

def separate_audio(input_path, output_path):
    separator = Separator('spleeter:2stems')
    separator.separate_to_file(input_path, output_path)

if __name__ == '__main__':
    if len(sys.argv) != 3:
        print("Usage: python spleeter_separation.py <input_path> <output_path>")
        sys.exit(1)
    input_path = sys.argv[1]
    output_path = sys.argv[2]
    separate_audio(input_path, output_path)

File: server/utils/spleeter_separation.js
const { exec } = require('child_process');
const path = require('path');

async function separateAudio(inputPath, outputPath) {
  return new Promise((resolve, reject) => {
    const command = `python ${path.resolve(__dirname, 'spleeter_separation.py')} ${inputPath} ${outputPath}`;
    exec(command, (error, stdout, stderr) => {
      if (error) {
        console.error(`Error separating audio: ${stderr}`);
        return reject(new Error(`Error separating audio: ${stderr}`));
      }
      console.log(`Audio separated: ${stdout}`);
      resolve(stdout);
    });
  });
}

module.exports = { separateAudio };
